From 08551b891e74445c6d7093edc1db54a819d7268b Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Thu, 9 Jan 2020 14:44:54 +0100
Subject: [PATCH 5/5] Use packaged version of mapbox-variant library.

---
 external/variant/CMakeLists.txt | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/external/variant/CMakeLists.txt b/external/variant/CMakeLists.txt
index 5b5beef..b0fb6cb 100644
--- a/external/variant/CMakeLists.txt
+++ b/external/variant/CMakeLists.txt
@@ -7,7 +7,12 @@
 add_library(external_variant INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_variant ALIAS external_variant)
 
-target_include_directories(external_variant SYSTEM
-INTERFACE
-    ${third_party_loc}/variant/include
-)
+if (DESKTOP_APP_USE_PACKAGED)
+    find_path(VARIANT_INCLUDE_DIRS mapbox/variant.hpp)
+    target_include_directories(external_variant INTERFACE ${VARIANT_INCLUDE_DIRS})
+else()
+    target_include_directories(external_variant SYSTEM
+    INTERFACE
+        ${third_party_loc}/variant/include
+    )
+endif()
-- 
2.24.1

