From 6c46194009a0bf26c1f383060a9d14378efa4ded Mon Sep 17 00:00:00 2001
From: Aleksey Kiporskiy <kiporscky@live.ru>
Date: Tue, 24 Mar 2020 13:58:39 +0500
Subject: [PATCH] bugfix: show chat description

---
 Telegram/SourceFiles/data/data_chat.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Telegram/SourceFiles/data/data_chat.cpp b/Telegram/SourceFiles/data/data_chat.cpp
index 9e7870b8ea..d3dfbc9e9c 100644
--- a/Telegram/SourceFiles/data/data_chat.cpp
+++ b/Telegram/SourceFiles/data/data_chat.cpp
@@ -350,6 +350,7 @@ void ApplyChatUpdate(not_null<ChatData*> chat, const MTPDchatFull &update) {
 	}
 	chat->checkFolder(update.vfolder_id().value_or_empty());
 	chat->fullUpdated();
+	chat->setAbout(qs(update.vabout()));
 
 	chat->session().api().applyNotifySettings(
 		MTP_inputNotifyPeer(chat->input),
