From 536d41410cfedc6d28fc7a4a4eed86bfcfce0cf4 Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Sat, 6 Jun 2020 20:21:32 +0200
Subject: [PATCH] Correctly specified required version of the GSL library.

Signed-off-by: Vitaly Zaitsev <vitaly@easycoding.org>
---
 external/gsl/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/external/gsl/CMakeLists.txt b/cmake/external/gsl/CMakeLists.txt
index 312447c..8762b36 100644
--- a/cmake/external/gsl/CMakeLists.txt
+++ b/cmake/external/gsl/CMakeLists.txt
@@ -8,7 +8,7 @@ add_library(external_gsl INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_gsl ALIAS external_gsl)
 
 if (DESKTOP_APP_USE_PACKAGED_GSL AND NOT DESKTOP_APP_USE_PACKAGED_LAZY)
-    find_package(Microsoft.GSL CONFIG REQUIRED)
+    find_package(Microsoft.GSL 3.1.0 CONFIG REQUIRED)
     target_link_libraries(external_gsl INTERFACE Microsoft.GSL::GSL)
 else()
     target_include_directories(external_gsl SYSTEM
-- 
2.26.2

